src/constructs/static-site/static-site.ts

```diff
@@ -0,0 +1,322 @@
+        description: `Canonical user ID used by CloudFront distribution. This should be granted 's3:GetObject' access to S3 bucket '${preExistingOriginBucketName}'.`,
+        value: cloudFrontOriginAccessIdentityS3CanonicalUserId,
+      });
+      Annotations.of(scope).addWarning(
+        `Origin bucket not managed by this stack. Grant 's3:GetObject' permission on ${preExistingOriginBucketName} to the principal listed as an Output on this stack. See https://github.com/awsdocs/amazon-s3-userguide/blob/d9fa2605ba253b3ca6ca1a83da5fa6b0b9b6debb/doc_source/s3-bucket-user-policy-specifying-principal-intro.md.`
+      );
+    } else {
+      bucket.addToResourcePolicy(
+        new PolicyStatement({
+          actions: ["s3:GetObject"],
+          resources: [bucket.arnForObjects(`${scope.stage}/${app}/*`)],
+          principals: [new CanonicalUserPrincipal(cloudFrontOriginAccessIdentityS3CanonicalUserId)],
+        })
+      );
+
+      new CfnOutput(scope, AppIdentity.suffixText({ app }, "OriginBucketName"), {
+        description: "S3 bucket origin for CloudFront.",
+        value: bucket.bucketName,
+      });
+    }
+
+    return bucket;
+  }
+
+  constructor(scope: GuStack, id: string, props: GuStaticSiteProps) {
+    const {
+      app,
+      domainNameProps,
+      behaviours = [
+        {
+          compress: true,
+          isDefaultBehavior: true,
+          viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        },
+      ],
+      withoutDns = false,
+    } = props;
+    const { stage } = scope;
+
+    const certificate = GuStaticSite.getCertificate(scope, props);
+
+    const domainName = StageAwareValue.isStageValue(domainNameProps)
+      ? scope.withStageDependentValue({
+          app,
+          variableName: GuCertificate.mappingVariableName,
+          stageValues: {
+            [Stage.CODE]: domainNameProps.CODE.domainName,
+            [Stage.PROD]: domainNameProps.PROD.domainName,
+          },
+        })
+      : domainNameProps.INFRA.domainName;
+
+    const viewerCertificate = ViewerCertificate.fromAcmCertificate(certificate, {
+      aliases: [domainName],
+    });
+
+    const originAccessIdentity = new OriginAccessIdentity(
+      scope,
+      AppIdentity.suffixText({ app }, "CloudFrontOriginAccessIdentity"),
+      {
+        comment: `Origin Access Identity for ${app} ${stage}`,
+      }
+    );
+
+    const s3Bucket = GuStaticSite.getBucket(scope, props, originAccessIdentity);
+
+    super(scope, id, {
+      ...props,
+      comment: `CloudFront distribution for ${app} ${stage}`,
+      viewerCertificate,
+      originConfigs: [
+        {
+          s3OriginSource: {
+            s3BucketSource: s3Bucket,
+            originAccessIdentity,
+            originPath: `/${stage}/${app}`,
+          },
+          behaviors: behaviours,
+        },
+      ],
+    });
+
+    if (!withoutDns) {
+      new GuCname(scope, AppIdentity.suffixText({ app }, "DnsRecord"), {
```

Comment:
We [decided *not* to create the `GuCname`](https://github.com/guardian/cdk/discussions/913) as part of our other patterns. I'm not sure if the same drawbacks apply here, or whether we want to follow the same approach for consistency?

Reply:
The only reason I opted to create it by default is we're already creating the CloudFront distribution and passing in the `domainNameProps`. Consistency would be better, however so will revisit this.